lib/Dialect/SV/SVOps.cpp

```diff
@@ -736,7 +736,7 @@ ParseResult CaseOp::parse(OpAsmParser &parser, OperationState &result) {
   StringRef keyword;
   if (!parser.parseOptionalKeyword(&keyword, {"case", "casex", "casez"})) {
     auto kind = symbolizeCaseStmtType(keyword);
-    auto caseEnum = static_cast<int32_t>(kind.getValue());
+    auto caseEnum = static_cast<int32_t>(kind.value());
```

Comment:
Glad you asked, thanks for reviewing this carefully!

Maybe it's not a great reason, but yes: per [the guidelines mentioned here (flang-based I think)](https://discourse.llvm.org/t/deprecating-llvm-optional-x-hasvalue-getvalue-getvalueor/63716) (which seemed reasonable, since not in our or LLVM's style guide otherwise), I tried to use `value()` if the presence check doesn't (clearly) dominate the access.  Here, the proof that this is okay relies on comparing the keyword array passed to `parseOptionalKeyword` with the supported cases in `symbolizeCastStmtType` in order to ensure this will have a valid `kind`.  While I'm satisfied I'm not sure it's obvious or that the two for sure won't drift.
By contrast, in SVOps.cpp (`parseEventList`) the symbolized kind is checked to be valid and so there it's just `*kind`.

Personally I kinda like the `operator bool` and `operator *` in most cases, but https://reviews.llvm.org/D128131 suggests it's somewhat controversial and does have some clarity drawbacks (especially if the underlying type is something you might normally convert to bool or dereference!).

Happy to do this however, especially if there's a consensus on what's preferred!

Reply:
That makes sense. This one particularly stood out because line 509 uses `*kind`. However, the "get value" is clearly dominated by a check.

This feels like one of those things that is destined to drift. 🥲 No idea what to do about that or if it's an actual concern. (I think not.)

Thanks for digging deep into the reasoning on the flang side of things. I do think it makes sense to default to that unless we have other strong opinions.